src/viam/sdk/services/mlmodel/server.cpp

```diff
@@ -34,184 +35,134 @@ void MLModelServiceServer::register_server(std::shared_ptr<Server> server) {
 ::grpc::Status MLModelServiceServer::Infer(
     ::grpc::ServerContext* context,
     const ::viam::service::mlmodel::v1::InferRequest* request,
-    ::viam::service::mlmodel::v1::InferResponse* response) noexcept try {
-    if (!request) {
-        return {::grpc::StatusCode::INVALID_ARGUMENT, "Called [Infer] without a request"};
-    };
-
-    std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
-
-    if (!rb) {
-        return {::grpc::UNKNOWN, "resource not found: " + request->name()};
-    }
-
-    std::shared_ptr<MLModelService> mlms = std::dynamic_pointer_cast<MLModelService>(rb);
-
-    if (!request->has_input_tensors()) {
-        return {::grpc::StatusCode::INVALID_ARGUMENT, "Called [Infer] with no inputs"};
-    }
-
-    AttributeMap extra;
-    if (request->has_extra()) {
-        extra = struct_to_map(request->extra());
-    }
-
-    const auto md = mlms->metadata(extra);
-    MLModelService::named_tensor_views inputs;
-    for (const auto& input : md.inputs) {
-        const auto where = request->input_tensors().tensors().find(input.name);
-        if (where == request->input_tensors().tensors().end()) {
-            // Ignore any inputs for which we don't have metadata, since
-            // we can't validate the type info.
-            //
-            // TODO: Should this be an error? For now we just don't decode
-            // those tensors.
-            continue;
+    ::viam::service::mlmodel::v1::InferResponse* response) noexcept {
+    return make_service_helper<MLModelService>(
+        "MLModelServiceServer::Infer", this, request)([&](auto& helper, auto& mlms) {
+        if (!request->has_input_tensors()) {
+            return helper.fail(::grpc::INVALID_ARGUMENT, "Called with no input tensors");
```

Comment:
Potentially, we could add new `fail` functions like `failInvalidArgument`, etc.

Reply:
Yeah `failInvalidArgument` in particular seems helpful. One for every [gRPC status code](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) could be cool but is probably overkill for now.